src/NumberTheory/QuadFormAndIsom/finite_group_actions.jl

```diff
@@ -0,0 +1,871 @@
+    return false
+  end
+
+  if ambient_representation
+    !is_isometry(ambient_space(L), f) && return false
+    ok, fL =  can_solve_with_solution(basis_matrix(L), basis_matrix(L)*f)
+  else
+    ok = true
+    fL = f
+  end
+  if !ok || !isone(denominator(fL))
+    return false
+  elseif !(rank(L) == nrows(fL) == ncols(fL))
+    return false
+  elseif fL*gram_matrix(L)*transpose(fL) != gram_matrix(L)
+    return false
+  elseif !is_stable
+    return true
+  end
+  fLQ = solve(basis_matrix(L), fL*basis_matrix(L); side=:right)
+  q = discriminant_group(L)
+  fq = hom(q, q, elem_type(q)[q(lift(t)*fLQ) for t in gens(q)])
+  return isone(matrix(fq))
+end
+
+@doc raw"""
+    is_isometry_list(
+      L::ZZLat,
+      V::Vector{QQMatrix},
+      ambient_representation::Bool = false;
+      is_special::Bool=false,
+      is_stable::Bool = false,
+    ) -> Bool
+
+Return whether the matrices in ``V`` define isometries of the lattice ``L``.
+
+If `is_special` is set to `true`, return whether all isometries in ``V``
+have determinant 1.
+
+If `is_stable` is set to `true`, return whether all isometries in ``V``
+act trivially on the discriminant group of ``L`` (note: this requires
+``L`` being integral).
+
+If `ambient_representation` is set to `true`, the matrices in ``V`` are
+considered as $\mathbb{Q}$-linear maps on the ambient space of ``L``.
+"""
+function is_isometry_list(
+    L::ZZLat,
+    V::Vector{QQMatrix},
+    ambient_representation::Bool = false;
+    is_special::Bool=false,
+    is_stable::Bool=false,
+  )
+  return all(f -> is_isometry(L, f, ambient_representation; is_special, is_stable), V)
+end
+
+@doc raw"""
+    is_isometry_group(
+      L::ZZLat,
+      G::MatrixGroup,
+      ambient_representation::Bool = false;
+      is_special::Bool=false,
+      is_stable::Bool=false,
+    ) -> Bool
+
+Return whether the matrix ``G`` defines a group of isometries of the lattice
+``L``.
+
+If `is_special` is set to `true`, return whether all isometries in ``G``
+have determinant 1.
+
+If `is_stable` is set to `true`, return whether all isometries in ``G``
+act trivially on the discriminant group of ``L`` (note: this requires
+``L`` being integral).
+
+If `ambient_representation` is set to `true`, the group ``G`` is considered as
+a group of $\mathbb{Q}$-linear automorphisms on the ambient space of ``L``.
+"""
+function is_isometry_group(
+    L::ZZLat,
+    G::MatrixGroup,
+    ambient_representation::Bool = false;
+    is_special::Bool=false,
+    is_stable::Bool=false,
+  )
+  return all(f -> is_isometry(L, matrix(f), ambient_representation; is_special, is_stable), gens(G))
+end
+
+###############################################################################
+#
+#  Change of representation
+#
+###############################################################################
+
+###############################################################################
+# Lattice to ambient
+
+@doc raw"""
+    representation_in_ambient_coordinates(
```

Comment:
I would have called it like this:
```suggestion
    extend_to_ambient_space(
```